modules/gapi/src/compiler/passes/pattern_to_substitute_matching.cpp

```diff
@@ -0,0 +1,176 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019 Intel Corporation
+
+#include "pattern_matching.hpp"
+
+#include <ade/util/zip_range.hpp>
+
+#define UNUSED(x) (void)x
+
+namespace cv { namespace gimpl {
+
+namespace {
+using Graph = GModel::Graph;
+using Metadata = typename Graph::CMetadataT;
+using VisitedMatchings = std::list<std::pair<ade::NodeHandle, ade::NodeHandle>>;
+
+// Returns true if two DATA nodes are semantically and structurally identical:
+//  - both nodes have the same GShape
+//  - both nodes have the same storage
+//  - both nodes have the same meta
```

Comment:
Should be removed, since you check shape and storage only

Reply:
Updated this part, thanks!